packages/runtime/src/useSSRComputation_Client.tsx

```diff
@@ -18,25 +18,27 @@ export default function useSSRComputation_Client(importFn: () => Promise<{ defau
     let isMounted = true;
     importFn().then((module) => {
       if (!isMounted) return;
-      // Wrapping to an empty function to avoid calling the function immediately.
-      // https://medium.com/swlh/how-to-store-a-function-with-the-usestate-hook-in-react-8a88dd4eede1
-      setFn(() => module.default);
+      const result = module.default();
+      if (result && typeof result.then === 'function') {
+        result.then(asyncResult => {
+          if (!isMounted) return;
+          cache[cacheKey] = asyncResult;
+          forceUpdate((x) => x + 1);
+        });
+      } else {
+        cache[cacheKey] = result;
+        forceUpdate((x) => x + 1);
```

Comment:
Should also check `if (!isMounted) return;`

Reply:
Updated the logic to prevent loading the function again if the parameters only changed without changing the file path